platform/ui/src/views/StudyList/components/StudyListFilter.js

```diff
@@ -108,61 +59,43 @@ const StudyListFilter = ({
     }
   };
 
-  const handleFilterValueChange = (event, name) => {
-    const { value } = event.target;
-    setcurrentFiltersValues(prevState => ({
-      ...prevState,
-      [name]: value,
-    }));
-  };
-
   const clearFilters = () => {
-    setcurrentFiltersValues(defaultProps.filtersValues);
+    setcurrentFiltersValues(filtersValues);
   };
 
-  const isFiltering = () => {
-    return Object.keys(currentFiltersValues).some(name => {
-      const filterValue = currentFiltersValues[name];
-      return filterValue !== defaultProps.filtersValues[name];
-    });
-  };
-
-  const renderInput = (inputType, name, { selectOptions }) => {
-    switch (inputType) {
-      case 'date-range': {
-        return (
-          <div className="relative">
-            <DateRange
-              startDate={currentFiltersValues.startDate}
-              endDate={currentFiltersValues.endDate}
-              onChange={({ startDate, endDate, preset = false }) => {
-                setcurrentFiltersValues(state => ({
-                  ...state,
-                  startDate,
-                  endDate,
-                }));
-              }}
-            />
-          </div>
-        );
+  const renderFieldInputComponent = ({
+    name,
+    displayName,
+    inputProps,
+    isSortable,
+    inputType,
+  }) => {
+    const _isSortable = isSortable && numOfStudies <= 100 && numOfStudies > 0;
+    const _sortDirection = sortBy !== name ? 'none' : sortDirection;
+    const onLabelClick = () => {
+      if (_isSortable) {
+        handleFilterLabelClick(name);
       }
-      case 'select': {
-        return <Select options={selectOptions}></Select>;
-      }
-      case 'text': {
-        return (
-          <Input
-            className="border-custom-blue mt-2 bg-black"
-            type="text"
-            containerClassName="mr-2"
-            value={currentFiltersValues[name] || ''}
-            onChange={event => handleFilterValueChange(event, name)}
-          />
-        );
-      }
-      default:
-        break;
-    }
+    };
+    const FilterInputComponent = getFieldInputComponent(inputType);
+
+    return (
+      <FilterInputComponent
+        key={name}
+        label={displayName}
+        isSortable={_isSortable}
+        sortDirection={_sortDirection}
+        onLabelClick={onLabelClick}
+        {...inputProps}
```

Comment:
`inputProps` is currently used for the Select options, right? As we should avoid spreading props, isn't it better to explicitly pass this value to an `options` prop?

Reply:
The code will be a little bit bigger now, but I removed this spreading porps.